### PR TITLE
Log warning for variable values exceeding 32KB

### LINKED ISSUE #3891: Show warning when a variable bigger then 32766 bytes is created
**Is your feature request related to a problem? Please describe.**

Variables bigger than 32766 bytes cannot be exported to elasticsearch with the default configuration (#32766). Also we want to discourage users from create big variables in general. 

**Describe the solution you'd like**

Every time a variable that is bigger then 32766 bytes is created a warning is printed to the Zeebe log.

**Describe alternatives you've considered**

- Build it into the clients but that would not cover input mappings etc., also we don't have control over all clients

### PR DESCRIPTION
**Pull request by menski**
## Description

To notify users that we discourage usage of big variable values we log a warning for variables which exceed 32KB

## Related issues

closes #3891

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Review by npepinpe — COMMENTED** [review]
I was under the impression this was a limit purely related to ES, so I thought we'd log it in the exporter itself. What is the gain of logging this in general (e.g. even without ES enabled)? Is it complicated to do it only there?

**Comment by menski** [comment]
I would argue that this warning, in general, is useful as we don't want large variables in the system. I also don't know how other systems behave if our variables get too big. But if you are not comfortable with the message I can try to make it depend on if ES exporter is enabled.

**Review by npepinpe — COMMENTED** [review]
I forgot what we decided in the end - should we move this to exporter-only or keep it for all variables? On one hand we should discourage users to use large variables - on the other hand we don't offer alternatives, and if they want to this warning quickly becomes overwhelming.

**Comment by menski** [comment]
Our result was that I understood your point and wanted to check how to only show the message if the es exporter is enabled or to add a configuration to let the user configure the max size. But I currently struggle with how to get the information/configuration to the variable state. Let's discuss this tomorrow in face to face.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: b2d4aa73cdc8971d6df445d1e42d9537f638c55b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by menski** [comment]
@npepinpe I change that the log message is generated in the elasticsearch exporter as I didn't wanted to push the information if the es exporter is configured to the variable state. But I don't know how to test it, I tried with the `MockRecord` but that fails as I have to case the record value to get the variable value out of it. Maybe you have an idea?

**Comment by npepinpe** [comment]
I'm not sure I understand the issue, just interrupt me whenever and let's quickly check it out.

**Comment by menski** [comment]
@npepinpe I wrote a test case using mocks, please have a look again

File: exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ElasticsearchClientTest {
+
+  public static final long RECORD_KEY = 1234L;
```
Comment (npepinpe) [inline exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchClientTest.java]:
```suggestion
  private static final long RECORD_KEY = 1234L;
```

**Review by npepinpe — APPROVED** [review]
:+1:

**Pull request merged**